### ISSUE TITLE
numEntries parameter is ignored when results are previously cached

### ISSUE THREAD

**Issue by AndyNewmanCRIBL** [paulsena/Cribl-Log-Collector-Interview#3]

**Comment by AndyNewmanCRIBL** [comment]
```
$ curl -u cribl:password "http://localhost:8080/cribl/log/tail?filename=test.txt" | jq
{
  "logEntries": [
    "Log line 7",
    "Log line 6",
    "Log line 5",
    "Log line 4",
    "Log line 3",
    "Log line 2",
    "Log line 1"
  ],
  "filterUsed": null
}
andy@Andy-laptop ~/src/interview/Cribl-Log-Collector-Interview/logs (main) $ curl -u cribl:password "http://localhost:8080/cribl/log/tail?filename=test.txt&numEntries=2" | jq

{
  "logEntries": [
    "Log line 7",
    "Log line 6",
    "Log line 5",
    "Log line 4",
    "Log line 3",
    "Log line 2",
    "Log line 1"
  ],
  "filterUsed": null
}
andy@Andy-laptop ~/src/interview/Cribl-Log-Collector-Interview/logs (main) $ 
```

the first request is without numEntries.  The second request sets numEntries to 3, which is ignored

**Comment by paulsena** [comment]
Thank you for your feedback. Bug fixed in: [e58c7e82349721fd9c5ef013f9592217905866c7](https://github.com/paulsena/Cribl-Log-Collector-Interview/commit/e58c7e82349721fd9c5ef013f9592217905866c7)

```
Fixed caching of results with numEntries request parameter. Checks cached list count is below request line number, if not, request the file watcher to run again.
  If below, use previous cached value and limits list count to requested. Caveat here to get efficient caching is that we limit before we filter.
  ```
